File: sp48/crypt/prohorow/test2.c
/****************************************************************************/
/*                                                                          */
/*      Data Encryption ToolKit. Version 1.3                                */
/*                                                                          */
/*      test2.c -- test module for DES module 2                             */
/*                                                                          */
/*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    */
/*                                                                          */
/*      Purpose:                                                            */
/*        Test encryption_key and decryption_key functions.                 */
/*                                                                          */
/****************************************************************************/


#include <stdio.h>
#include "encrypt.h"

int test_value[4] = {1, 2, 3, 4};
char key[] = "Prokhorow";
const int KEY_NUMBER = 1;
char key_area[ENCRYPTION_KEY_AREA_SIZE (KEY_NUMBER)];

void test_write () {
  for (int i = 0; i < 4; i ++)
    printf ("%i ", test_value[i]);
  printf ("\n");
  }

void main () {
  test_write ();
  encryption_key (key_area, key);
  encrypt_block (key_area, test_value);
  test_write ();
  decryption_key (key_area, key);
  encrypt_block (key_area, test_value);
  test_write ();
  }

File: sp48/crypt/prohorow/test2.pas
{****************************************************************************}
{*                                                                          *}
{*      Data Encryption ToolKit. Version 1.3                                *}
{*                                                                          *}
{*      test2.pas -- test module for DES module 2                           *}
{*                                                                          *}
{*      Copyright (c) 1991, 1992, Andrew Prokhorow. All rights reserved.    *}
{*                                                                          *}
{*      Purpose:                                                            *}
{*        Test EncryptionKey and DecryptionKey procedures.                  *}
{*                                                                          *}
{****************************************************************************}


program Test2;

  uses Encrypt;

  const
    TestValue: array [1..4] of Integer = (1, 2, 3, 4);
    Key: String = 'Prokhorow';
    EncryptionKeyNumber = 1;

{$i encrcon.inc}

  var
    KeyArea: array [1..EncryptionKeyAreaSize] of Char;

  procedure TestWrite;

    var
      I: Integer;

    begin
      for I := 1 to 4 do
        Write (TestValue[I], ' ');
      WriteLn
    end;

  begin
    TestWrite;
    EncryptionKey (KeyArea, Key);
    EncryptBlock (KeyArea, TestValue);
    TestWrite;
    DecryptionKey (KeyArea, Key);
    EncryptBlock (KeyArea, TestValue);
    TestWrite
  end.
